src/ts/ConvivaAnalytics.ts

```
@@ -657,79 +698,127 @@ export class ConvivaAnalytics {
+
+    if (bitrateKbps) {
+      this.debugLog('[ ConvivaAdAnalytics ] report ad bitrate', bitrateKbps);
+      this.convivaAdAnalytics.reportAdMetric(Conviva.Constants.Playback.BITRATE, bitrateKbps);
     }
+  }
+
+  // Fires before `onAdBreakFinished`
+  private onAdFinished = (event: AdEvent) => {
+    this.debugLog('[ Player Event ] ad finished', event);
+
+    this.debugLog('[ ConvivaAdAnalytics ] report ad ended', {
+      event,
+    });
+    this.convivaAdAnalytics.reportAdEnded();
+  }
 
+  // Fires before `onAdBreakFinished`
+  private onAdSkipped = (event: AdEvent) => {
+    this.debugLog('[ Player Event ] ad skipped', event);
+
+    this.debugLog('[ ConvivaAdAnalytics ] report ad skipped', event);
+    this.convivaAdAnalytics.reportAdSkipped();
+
+    this.onCustomEvent(event);
+  };
+
+  private onAdBreakFinished = (event: AdBreakEvent | ErrorEvent) => {
+    this.debugLog('[ Player Event ] adbreak finished', event);
+    this.isAdBreak = false;
+
+    this.debugLog('[ ConvivaAnalytics ] report ad break ended', event);
     this.convivaVideoAnalytics.reportAdBreakEnded();
+
+    this.debugLog('[ ConvivaAnalytics ] report playing playback state');
     this.convivaVideoAnalytics.reportPlaybackMetric(
       Conviva.Constants.Playback.PLAYER_STATE,
       Conviva.Constants.PlayerState.PLAYING,
     );
   };
 
-  private onAdSkipped = (event: AdEvent) => {
-    this.onCustomEvent(event);
-  };
+  private onAdError = (event: ErrorEvent) => {
+    this.debugLog('[ Player Event ] ad error', event);
+
+    const formattedError = AdHelper.formatAdErrorEvent(event);
+
+    this.debugLog('[ ConvivaAdAnalytics ] report ad error', {
+      event,
+      formattedError,
+    });
+    this.convivaAdAnalytics.reportAdError(formattedError, Conviva.Constants.ErrorSeverity.WARNING);
 
-  private onAdError = (event: AdEvent) => {
     this.onCustomEvent(event);
```

Same as above. Do we still need to report the error as custom event now we have the AdAnalytics? 🤔

---

As I mentioned, I would keep it at least for some time.